python/src/opendp/mod.py

```diff
@@ -608,6 +617,128 @@ def __iter__(self):
         raise ValueError("Domain does not support iteration")
 
 
+class AtomDomain(Domain):
+    '''`AtomDomain` is the domain of all values of an atomic type `T`.
+
+    If bounds are set, then the domain is restricted to the bounds.
+    If nullable is set, then null value(s) are included in the domain.
+    '''
+
+    _type_ = AnyDomain
+    
+    @property
+    def bounds(self) -> tuple[float, float] | None:
+        '''Bounds of the domain, if they exist'''
+        from opendp.domains import _atom_domain_get_bounds_closed
+        return _atom_domain_get_bounds_closed(self)
+    
+    @property
+    def nullable(self) -> bool:
+        '''Whether the domain includes null values'''
+        from opendp.domains import _atom_domain_get_nullable
+        return _atom_domain_get_nullable(self)
+    
+    def __del__(self):
+        # this is to avoid a double-free, as the Domain.__del__ will, suprisingly, still be called
+        pass
+
+class OptionDomain(Domain):
+    '''A domain that represents nullity via the Option type.
```

Comment:
Concretely, what is the "Option type"? Maybe rephrase:
```suggestion
    '''A domain whose members are either members of the ``element_domain``, or ``None``.
```

Reply:
and add a link to the real constructor?